CarRentals/CarRentalsWebAPI/Controllers/RentalsController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using CarRentals.Models;
+using CarRentalsWebAPI.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CarRentalsWebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class RentalsController : ControllerBase
+    {
+        private readonly CarRentalsContext _context;
+
+        public RentalsController(CarRentalsContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Rentals
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Rental>>> GetRentals()
+        {
+            return await _context.Rentals.ToListAsync();
+        }
+
+        // GET: api/Rentals/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Rental>> GetRental(int id)
+        {
+            var rental = await _context.Rentals.FindAsync(id);
+
+            if (rental == null)
+            {
+                return NotFound();
+            }
+
+            return rental;
+        }
+
+        // PUT: api/Rentals/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutRental(int id, Rental rental)
+        {
+            if (id != rental.Id)
+            {
+                return BadRequest();
+            }
```

Comment:
this is not the usual way (I'm not saying this is wrong), maybe instead of returning a bad request you can just assign the Id that was passed as a discrete parameter? something like 
```csharp
rental.Id = id;
```